geth/common/notification.go

```diff
@@ -0,0 +1,16 @@
+package common
```

Comment:
This is useful practice to prevent cyclic dependencies. But i agree that "common" package for interfaces looks strange and uninformative. We can do this refactoring in separete issue.

Reply:
> This is useful practice to prevent cyclic dependencies.

Right, but I doubt it's needed in most of the cases here.

> We can do this refactoring in separete issue.

Agree. It's better to be consistent for now.

Reply:
I've added better interfaces, more tests and rewrite comments in the last commits. Could you review PR?